src/item_pocket.cpp

```diff
@@ -1706,6 +1706,54 @@ void item_pocket::process( map &here, Character *carrier, const tripoint &pos, f
     }
 }
 
+bool item_pocket::leak( map &here, Character *carrier, const tripoint &pos, item_pocket *pocke )
+{
+    std::vector<item *> erases;
+    for( auto iter = contents.begin(); iter != contents.end(); ) {
+        if( iter->leak( here, carrier, pos, this ) ) {
+            if( watertight() ) {
+                continue;
```

Comment:
Shouldn't there be a `iter++` here? This continue currently leads to an infinite loop.

Reply:
Exactly, thank you.
I will fix it.